client/src/views/users/GreetUserModal.vue

```diff
@@ -24,28 +24,30 @@
         v-if="isLargeDisplay"
       >
         <ion-title class="modal-header__title title-h2">
-          {{ $msTranslate(getStep(pageStep).title) }}
+          {{ $msTranslate(steps[pageStep]?.title) }}
         </ion-title>
         <ion-text
-          v-if="getStep(pageStep).subtitle"
           class="modal-header__text body"
+          v-if="steps[pageStep]?.subtitle"
         >
-          {{ $msTranslate(getStep(pageStep).subtitle) }}
+          {{ $msTranslate(steps[pageStep]?.subtitle) }}
         </ion-text>
       </ion-header>
-      <small-display-step-modal-header
-        v-else
-        @close-clicked="cancelModal()"
-        :title="getStep(pageStep).title"
-        :subtitle="getStep(pageStep).subtitle"
-        :step="{
-          icon: personAdd,
-          title: 'UsersPage.greet.titles.waitForGuest',
-          current: getStep(pageStep).currentStep,
-          total: 4,
-        }"
-      />
-
+      <template v-else>
+        <small-display-modal-header
+          v-if="pageStep === GreetUserStep.WaitForGuest || pageStep === GreetUserStep.Summary"
+          @close-clicked="cancelModal()"
+          :title="steps[pageStep]?.title"
```

Comment:
When on `GreetUserStep.Summary` the index goes out of range of the array since the steps values are numbered 1 to 6 instead of 0 to 5.